src/Data/Nat/Base.agda

```diff
@@ -130,6 +132,47 @@ instance
 >-nonZero⁻¹ : ∀ n → .{{NonZero n}} → n > 0
 >-nonZero⁻¹ (suc n) = z<s
 
+-- The property of being a non-zero, non-unit
+
+pattern 2+ n = suc (suc n)
+
+trivial : ℕ → Bool
+trivial 0      = true
+trivial 1      = true
+trivial (2+ _) = false
```

Comment:
It's kind of weird to define predicates using Bool first, as we try and avoid using raw Bool wherever possible. We certainly wouldn't encourage the user to use such a function. Usually we define such predicates as records, e.g. `NonZero`?

Reply:
Point taken regarding not letting users have access to this.

Regarding the use of `record`s to encode such predicates, I've argued [on #2092](https://github.com/agda/agda-stdlib/issues/2092#issuecomment-1719250454)  that the additional indirection given by an explicit top-level record is overkill for what are in the end, instances of `T _`, and hence *either* trivially inhabited by `_` (avoiding bothering to mention `tt` explicitly), *else* trivially uninhabited and matched by `()`. (See also my comments on reducing all `Decidable` properties to `T?` via `Reflects.fromEquivalence`, and its weaponsation in #2149)

So I've been developing a style of first introducing the `Bool`-valued indicator/characteristic function, and then the predicate via `T ∘_`... but clearly the/one way to satisfy us both is to make `trivial` local to the definition of `NonTrivial`?

Reply:
Okay, so I've remembered the real reason they're all defined as records. As @gallais reminds us again and again, Agda behaves far far better when reconstructing meta-variables when the original definition is a record rather than a plain alias. 

I'm afraid I don't have good intuition why, but it's definitely true. In particular when I change the definition to a record, the bug where you need to redefine instances at the top of the `Primality` file goes away again. See the latest commit. 

So I'm pushing the change, and adding some documentation to `Data.Nat.Base`.